components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,171 @@
+Name:            kyma-default
+Resource         Used       Hard
+--------         ---        ---
+limits.memory    7348440Ki  10Gi
+requests.memory  3727576Ki  7Gi
+```
+
+The Console calls for the ResourceQuota status. The calls are triggered after: 
+- switching the environment
+- uploading a resource
+- creating a lambda
+- opening environment's details
+
+There are a few ways to check if the ResourceQuota exceeds its limit.
+
+## Overview
+ 
+The idea is to create an endpoint which returns the ResourceQuotaStatus.
+ResourceQuotaStatus contains the flag and the optional message. The value of this flag informs you if the ResourceQuota is exceeded.
+
+### Multistage check
+
+To check if you exceeded the ResourceQuota in the Environment you must implement a few steps.
+
+### Check the status of the ResourceQuotas
+
+List ResourceQuotas in the given Namespace and loop through them. You exceeded the ResourceQuota if at least one of the  `.status.used` values equals or is bigger than its equivalent from `.spec.hard`. 
+In this situation return ResourceQuotaStatus with a flag set to true.
+
+You need to specify which resources do you want to check.
+In the Kyma, we will check the following resources:
+1. Memory requests and limits
+2. Cpu requests and limits
+3. Number of Pods
+
+### Check ReplicaSets
+
+List ReplicaSets in the given Namespace and loop through them. You must check every ReplicaSet which number of desired replicas is not reached.
+First of all check if the ReplicaSet has an OwnerReference to the Deployment. If yes, you must get the Deployment's field `.spec.strategy.rollingUpdate.maxUnavailable` value.
+Parse that value to **int** and use it to decrease the number of desired replicas in the `if` statement which checks if the number of desired replicas is reached.
+
+>**NOTE** The value of `.spec.strategy.rollingUpdate.maxUnavailable` can be expressed as a percentage or a number. You can find more information [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable)
+
+If the desired number of replicas is not reached, then check if any ResourceQuota is not blocking the ReplicaSet from progressing.
+To achieve that implement logic which calculates if the ReplicaSet has enough memory to create the next replica.
+To calculate how much resources does the ReplicaSet needs to progress you must sum up the resource usage of all containers in the replica Pod. You also must calculate the `difference` from `.spec.hard` and `.status.used`. It will give you the total amount of available resources to use in the given Namespace.
+Loop through ResourceQuota list and check if any ResourceQuota is not blocking the ReplicaSet from progress by comparing available resources left in the Environment and the required resources to create next replica.
+If there is no enough resources to create the next replica and the number of desired replicas is not reached, then you returns a ResourceQuotaStatus with a flag set to true.
```

Comment:
1. `not enough`
2. ~then you returns~ -> `return`
3. ``with a flag set to `true`.``

Reply:
Done